src/main/java/com/spotify/confidence/ConfidenceValue.java

```diff
@@ -96,32 +98,70 @@ public java.util.List<ConfidenceValue> asList() {
     throw new IllegalStateException("Not a ListValue");
   }
 
-  public static ConfidenceValue of(int value) {
+  public static ConfidenceValue<java.lang.Integer> of(int value) {
     return new Integer(value);
   }
 
-  public static ConfidenceValue of(double value) {
+  public static ConfidenceValue<java.lang.Double> of(double value) {
     return new Double(value);
   }
 
-  public static ConfidenceValue of(Instant value) {
+  public static ConfidenceValue<Instant> of(Instant value) {
     return new Timestamp(value);
   }
 
-  public static ConfidenceValue of(LocalDate date) {
+  public static ConfidenceValue<LocalDate> of(LocalDate date) {
     return new Date(date);
   }
 
-  public static ConfidenceValue of(String value) {
+  public static ConfidenceValue<String> of(String value) {
     return new StringValue(value);
   }
 
-  public static ConfidenceValue of(boolean value) {
+  public static ConfidenceValue<Boolean> of(boolean value) {
     return new BooleanValue(value);
   }
 
-  public static List of(java.util.List<ConfidenceValue> values) {
-    return new List(values);
+  public static ConfidenceValue.List<String> ofStrings(
```

Comment:
One big improvement to this API would be if it could also take a `vararg` so you could call it like
```
final ConfidenceValue.List<String> listValue =
        ConfidenceValue.ofStrings(ConfidenceValue.of("item1"), ConfidenceValue.of("item2")));
```

But we can build that later if we feel like it.

Reply:
<img width="711" alt="Screenshot 2024-03-19 at 18 41 35" src="https://github.com/spotify/confidence-openfeature-provider-java/assets/1221595/968c9ff1-46c7-48cc-bec5-9451d61a4115">
I tried this thing out quickly but only achieved this which is not really great so lets drop it for now.